server/resp/src/main/java/org/infinispan/server/resp/Resp3AuthHandler.java

```diff
@@ -5,14 +5,17 @@
 import java.util.concurrent.CompletionStage;
 
 import javax.security.auth.Subject;
+import javax.security.sasl.SaslException;
 
 import org.infinispan.commons.util.concurrent.CompletableFutures;
-import org.infinispan.server.core.security.UsernamePasswordAuthenticator;
+import org.infinispan.server.resp.authentication.RespAuthenticator;
 import org.infinispan.server.resp.commands.AuthResp3Command;
 
 import io.netty.channel.ChannelHandlerContext;
+import io.netty.util.AttributeKey;
 
 public class Resp3AuthHandler extends CacheRespRequestHandler {
+   static final AttributeKey<Subject> SUBJECT_ATTRIBUTE_KEY = AttributeKey.newInstance("subject");
```

Comment:
@jabolina I'm adding the Subject to the `ConnectionMetadata` in https://github.com/infinispan/infinispan/pull/10964